stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/NextTextRecord.java

```diff
@@ -49,24 +59,77 @@
      */
     private static final int TXT_RECORD_CHANNEL_INDEX = 0;
 
+    /**
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(NextTextRecord.class);
+
+    /*
+     * Response obtained after reading record.
+     *
+     * @param result the result obtained after processing the record.
+     * @return the response obtained after reading record.
+     */
+/*
+    private static EventResult readRecordResponse(EventResult<String[], EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        String[] fields = result.getResponse();
+        return result;
+    }
+*/
+
+    /**
+     * Reads bytes asynchronously.
+     *
+     * @param recordChannel channel the bytes should be read from.
+     * @param context       event context.
+     * @return the fields which were read.
+     * @throws ExecutionException   errors which occur during execution.
+     * @throws InterruptedException during interrupt error.
+     */
+    private String[] readRecord(DelimitedRecordChannel recordChannel, EventContext context) throws ExecutionException,
+            InterruptedException {
+        DelimitedRecordReadEvent event = new DelimitedRecordReadEvent(recordChannel, context
+        );
+        CompletableFuture<EventResult> future = eventManager.publish(event);
+        //future.thenApply(NextTextRecord::readRecordResponse);
```

Comment:
Shall we remove the commented line?

Reply:
This needs to be uncommented upon the availability of the async functions